framework/include/interfaces/Coupleable.h

```diff
@@ -240,6 +240,62 @@ class Coupleable
   std::vector<const ADVectorVariableValue *>
   adCoupledVectorValues(const std::string & var_name) const;
 
+  /**
+   * Returns value of a coupled array variable for a given tag
+   * @param var_name Name of coupled array variable
+   * @param tag vector tag ID
+   * @param comp Component number for vector of coupled variables
+   * @return Reference to a VariableValue for the coupled array variable
+   * @see Kernel::_u
+   */
+  virtual const ArrayVariableValue &
+  coupledVectorTagArrayValue(const std::string & var_name, TagID tag, unsigned int comp = 0) const;
+
+  virtual const ArrayVariableValue & coupledVectorTagArrayValue(const std::string & var_name,
+                                                                const std::string & tag_name,
+                                                                unsigned int comp = 0) const;
+
+  /**
+   * Returns the values for all of a coupled variable's components for a given tag
+   * @param var_name Name of array coupled variable
+   * @param tag vector tag ID
+   * @return Vector of ArrayVariableValue pointers for each component of \p var_name
+   */
+  std::vector<const ArrayVariableValue *> coupledVectorTagArrayValues(const std::string & var_name,
```

Comment:
this should just be arrayvariablevalue ??

Reply:
Here coupledVectorTagArrayValue`s` returns multiple ArrayVariableValue variables - Note that var_name is the list of array variable names.